### PR TITLE
refactor(sbb-train-formation): lit migration

### PR DESCRIPTION
**Pull request by dauriamarco**
Migrate `sbb-train`, `sbb-train-wagon`, `sbb-train-blocked-passage` and `sbb-train-formation` to Lit.

### CODE REVIEW

File: src/components/sbb-icon/sbb-icon.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class SbbIcon extends LitElement {
   }
 
   protected override render(): TemplateResult {
-    setAttribute(this, 'role', 'img');
+    setAttribute(this, 'role', this.getAttribute('role') ?? 'img');
```
Comment (TomMenga) [inline src/components/sbb-icon/sbb-icon.tsx:110]: Try this, should work the same
```suggestion
    setAttribute(this, 'role', this.role ?? 'img');
```

**Review by TomMenga — CHANGES_REQUESTED** [review]
Nice work, some minor fixes

File: src/components/sbb-icon/sbb-icon.tsx (hunk 1 above)
Comment (dauriamarco) [reply src/components/sbb-icon/sbb-icon.tsx:110]: If I change this, the tests on CI will fail 🤷‍♂️

Comment (TomMenga) [reply src/components/sbb-icon/sbb-icon.tsx:110]: mmm maybe I'm missing something. Feel free to choose whether to debug it or leave it as it is

**Review by TomMenga — APPROVED** [review]

**Pull request merged**